enterprise/frontend/src/metabase-enterprise/google_drive/GdriveSyncStatus.tsx

```diff
@@ -4,103 +4,74 @@ import { match } from "ts-pattern";
-import { getUserIsAdmin } from "metabase/selectors/user";
 import StatusLarge from "metabase/status/components/StatusLarge";
 import { useGetGsheetsFolderQuery } from "metabase-enterprise/api";
-import type { DatabaseId, Settings } from "metabase-types/api";
+import { EnterpriseApi } from "metabase-enterprise/api/api";
+import type { DatabaseId, GdrivePayload } from "metabase-types/api";
 
-type GsheetsStatus = Settings["gsheets"]["status"];
+import { SYNC_POLL_INTERVAL } from "./constants";
+import { getStatus, useShowGdrive } from "./utils";
+
+type GsheetsStatus = GdrivePayload["status"];
 type ErrorPayload = { data?: { message: string } };
 
 export const GdriveSyncStatus = () => {
-  const gsheetsSetting = useSetting("gsheets");
-  const settingStatus = gsheetsSetting?.status;
-  const previousSettingStatus = usePrevious(settingStatus);
   const dispatch = useDispatch();
-  const isAdmin = useSelector(getUserIsAdmin);
-  const currentUser = useSelector(getCurrentUser);
-  const isCurrentAdmin =
-    isAdmin &&
-    gsheetsSetting?.["created-by-id"] &&
-    currentUser.id === gsheetsSetting?.["created-by-id"];
-  const [forceHide, setForceHide] = useState(
-    !isCurrentAdmin || settingStatus !== "loading",
-  );
-  const [syncError, setSyncError] = useState({ error: false, message: "" });
+  const showGdrive = useShowGdrive();
+
+  const [forceHide, setForceHide] = useState(true);
   const [dbId, setDbId] = useState<DatabaseId | undefined>();
 
-  const shouldPoll = isCurrentAdmin && settingStatus === "loading";
+  const res = useGetGsheetsFolderQuery(!showGdrive ? skipToken : undefined);
+  const { data: gdriveFolder, error: apiError } = res;
 
-  const { currentData: folderSync, error: apiError } = useGetGsheetsFolderQuery(
-    shouldPoll ? undefined : skipToken,
-    { pollingInterval: 3000 },
-  );
+  const currentUser = useSelector(getCurrentUser);
+  const isCurrentUser = currentUser?.id === gdriveFolder?.created_by_id;
+
+  const status = getStatus({ status: gdriveFolder?.status, error: apiError });
+
+  const previousStatus = usePrevious(status);
 
-  // if our polling endpoint changes away from loading, refresh the settings and show error, if any
   useEffect(() => {
-    if (folderSync?.status !== "loading" || apiError) {
-      if (apiError) {
-        console.error((apiError as ErrorPayload)?.data?.message);
-
-        setSyncError({
-          error: true,
-          // eslint-disable-next-line no-literal-metabase-strings -- admin UI
-          message: t`Please check that the folder is shared with the Metabase Service Account.`,
-        });
-      }
-
-      if (folderSync?.db_id) {
-        setDbId(folderSync.db_id);
-      }
-
-      dispatch(reloadSettings());
+    if (status === "syncing" && !forceHide) {
+      const timeout = setTimeout(() => {
+        dispatch(EnterpriseApi.util.invalidateTags(["gsheets-status"]));
+      }, SYNC_POLL_INTERVAL);
+      return () => {
+        clearTimeout(timeout);
+      };
     }
```

Comment:
Instead of using RTKQuery's built-in polling, this just sets up a timeout every time we get a `syncing` response to invalidate the query tag and trigger one refetch. This ends up be A LOT more predictable and reliable than the native polling approach.